app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java

```diff
@@ -228,4 +228,81 @@ public final boolean isDatabaseEmpty() {
+        String hash = conversionController.hashKey();
+        Block block = new Block(1, owner, hash, chainHash, senderHash, publicKey, iban, 0, false);
+        addBlock(block);
+        return block;
+    }
+
+    /**
+     * Create a block which adds a key for a certain user and weaves it into the blockchain.
+     * The initial trust value is zero.
+     * @param owner owner of the block
+     * @param contact of whom is the information
+     * @param publicKey public key you want to store
+     * @param iban IBAN number to store in this block
+     * @return the newly created block
+     * @throws Exception when the hashing algorithm is not available
+     */
+    public Block createKeyBlock(String owner, String contact, String publicKey, String iban) throws Exception {
+        return createBlock(owner, contact, publicKey, iban, false);
+    }
+
+    /**
+     * Create a block which revokes a key for a certain user and weaves it into the blockchain.
+     * The initial trust value is zero.
+     * @param owner owner of the block
+     * @param contact of whom is the information
+     * @param publicKey public key you want to store
+     * @param iban IBAN number to store in this block
+     * @return the newly created block
+     * @throws Exception when the hashing algorithm is not available
+     */
+    public Block createRevokeBlock(String owner, String contact, String publicKey, String iban) throws Exception {
+        return createBlock(owner, contact, publicKey, iban, true);
+    }
+
+    /**
+     * Creates a block with given revoke status. The block is added to the blockchain automatically
+     * with all fields set correctly.
+     * @param owner owner of the block
+     * @param contact of whom is the information
+     * @param publicKey public key you want to store
+     * @param iban IBAN number to store in this block
+     * @param revoke whether to revoke?
+     * @return the newly created block
+     * @throws Exception when the hashing algorithm is not available
+     */
+    private Block createBlock(String owner, String contact, String publicKey, String iban, boolean revoke) throws Exception {
+        Block latest = getLatestBlock(owner);
+        String previousBlockHash = latest.getOwnHash();
+
+        // always link to genesis of contact blocks
+        String contactBlockHash;
+        if (owner.equals(contact))
+            contactBlockHash = "N/A";
+        else
+            contactBlockHash = getBlocks(contact).get(0).getOwnHash();
+        int index = latest.getSequenceNumber() + 1;
+
+        ConversionController conversionController = new ConversionController(
+                owner, publicKey, previousBlockHash, contactBlockHash, iban
+        );
+        String hash = conversionController.hashKey();
+        Block block = new Block(index, owner, hash, previousBlockHash, contactBlockHash, publicKey, iban, 0, revoke);
```

Comment:
This must be done in another pull request, using a fresh TrustValues item that does not yet exist.